src/debugging/coreclr/LocalAspNetCoreSslManager.ts

```diff
@@ -0,0 +1,139 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import * as path from 'path';
+import { MessageItem } from 'vscode';
+import { ext } from '../../extensionVariables';
+import { PlatformOS } from '../../utils/platform';
+import { quickPickProjectFileItem } from '../../utils/quick-pick-file';
+import { quickPickWorkspaceFolder } from '../../utils/quickPickWorkspaceFolder';
+import { ProcessProvider } from './ChildProcessProvider';
+import { DotNetClient, TrustState } from './CommandLineDotNetClient';
+import { OSProvider } from './LocalOSProvider';
+import { NetCoreProjectProvider } from './netCoreProjectProvider';
+
+export type SecretsFolders = {
+    certificateFolder: string;
+    userSecretsFolder: string;
+}
+
+export interface AspNetCoreSslManager {
+    trustCertificateIfNecessary(): Promise<void>;
+    exportCertificateIfNecessary(projectFile: string | undefined, certificateExportPath: string | undefined): Promise<void>;
+    getHostSecretsFolders(): SecretsFolders;
+    getContainerSecretsFolders(platform: PlatformOS): SecretsFolders;
+}
+
+export class LocalAspNetCoreSslManager implements AspNetCoreSslManager {
+
+    private static _KnownConfiguredProjects: Set<string> = new Set<string>();
+    private static _CertificateTrustedOrSkipped: boolean = false;
+
+    constructor(
+        private readonly dotNetClient: DotNetClient,
+        private readonly netCoreProjectProvider: NetCoreProjectProvider,
+        private readonly processProvider: ProcessProvider,
+        private readonly osProvider: OSProvider) {
+    }
+
+    public async trustCertificateIfNecessary(): Promise<void> {
+        if (LocalAspNetCoreSslManager._CertificateTrustedOrSkipped) {
+            return;
+        }
+
+        const trusted = await this.dotNetClient.isCertificateTrusted();
+
+        if (trusted === TrustState.Trusted || trusted === TrustState.NotApplicable) {
+            LocalAspNetCoreSslManager._CertificateTrustedOrSkipped = true;
+            return;
+        }
+
+        if (this.osProvider.os === 'Windows') {
+            const trust: MessageItem = { title: 'Trust' };
+            const message = 'The ASP.NET Core HTTPS development certificate is not trusted. Your browser may show warnings about the site when opened. To trust the certificate manually, run \`dotnet dev-certs https --trust\`, or click "Trust" to run this command automatically. You may be prompted to trust the certificate.';

```

Comment:
Pretty long message - can we get a screenshot? And try to shorten? I feel like the purpose of a "Learn More" button is so that we don't have to explain every detail in the message haha

Reply:
Agreed, it is long. I wasn't super in love with it. Here it is on Windows (Mac loses the sentence about clicking "Trust"):

![image](https://user-images.githubusercontent.com/36966225/62388169-9be15300-b52a-11e9-92a1-35a9ce2f3c11.png)

I think we could get rid of the part about browser warning?

Reply:
Looks a little gross, but not as bad as I thought. Up to you

Reply:
I worked with the PM to shorten it,
![image](https://user-images.githubusercontent.com/36966225/62389819-d9e07600-b52e-11e9-9e55-4b466e89a480.png)